src/platforms/common/stm32/serialno.c

```diff
@@ -64,3 +72,32 @@ void read_serial_number(void)
 #endif
 	serial_no[DFU_SERIAL_LENGTH - 1] = '\0';
 }
+
+static char *utoa_upper(unsigned value, char *str, int base)
+{
+	const char digits[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+	/* Check base is supported. */
+	if ((base < 2) || (base > 36)) {
```

Comment:
Style nitpick, please drop the inner parens - `(base < 2 || base > 36)` - otherwise this turns off the GCC warning about accidental assignment

Reply:
Extra parens dropped, even if it's comparison, not assignment.